packages/grafana-ui/src/components/Layout/Box/Box.tsx

```diff
@@ -0,0 +1,214 @@
+import { css } from '@emotion/css';
+import React, { ElementType, useCallback } from 'react';
+
+import { GrafanaTheme2, ThemeSpacingTokens, ThemeShape, ThemeShadows } from '@grafana/data';
+
+import { useStyles2 } from '../../../themes';
+import { AlignItems, JustifyContent } from '../Flex/Flex';
+import { ResponsiveProp, getResponsiveStyle } from '../utils/responsiveness';
+
+type Display = 'flex' | 'block' | 'inline' | 'none';
+export type BackgroundColor = keyof GrafanaTheme2['colors']['background'] | 'error' | 'success' | 'warning' | 'info';
+export type BorderStyle = 'solid' | 'dashed';
+export type BorderColor = keyof GrafanaTheme2['colors']['border'] | 'error' | 'success' | 'warning' | 'info';
+export type BorderRadius = keyof ThemeShape['radius'];
+export type BoxShadow = keyof ThemeShadows;
+
+interface BoxProps {
+  // Margin props
+  /** Sets the property `margin` */
+  margin?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the properties `margin-top` and `margin-bottom`. Higher priority than margin. */
+  marginX?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the properties `margin-left` and `margin-right`. Higher priority than margin. */
+  marginY?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `margin-top`. Higher priority than margin and marginY. */
+  marginTop?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `margin-bottom`. Higher priority than margin and marginXY */
+  marginBottom?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `margin-left`. Higher priority than margin and marginX. */
+  marginLeft?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `margin-right`. Higher priority than margin and marginX. */
+  marginRight?: ResponsiveProp<ThemeSpacingTokens>;
+
+  // Padding props
+  /** Sets the property `padding` */
+  padding?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the properties `padding-top` and `padding-bottom`. Higher priority than padding. */
+  paddingX?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the properties `padding-left` and `padding-right`. Higher priority than padding. */
+  paddingY?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-top`. Higher priority than padding and paddingY. */
+  paddingTop?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-bottom`. Higher priority than padding and paddingY. */
+  paddingBottom?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-left`. Higher priority than padding and paddingX. */
+  paddingLeft?: ResponsiveProp<ThemeSpacingTokens>;
+  /** Sets the property `padding-right`. Higher priority than padding and paddingX. */
+  paddingRight?: ResponsiveProp<ThemeSpacingTokens>;
+
+  // Border Props
+  borderStyle?: ResponsiveProp<BorderStyle>;
+  borderColor?: ResponsiveProp<BorderColor>;
+  borderRadius?: ResponsiveProp<BorderRadius>;
```

Comment:
Should we also have a shorthand `border` property?

Reply:
My problem with the shorthand at the moment is that we don't have a token for `BorderWidth` (which we should), so we can't very strictly type it. Something we can add another time?